service/src/main/kotlin/fi/espoo/evaka/pis/PartnershipQueries.kt

```diff
@@ -116,51 +141,61 @@ fun Database.Transaction.createPartnership(
             fp1.end_date,
             fp1.conflict,
             ${aliasedPersonColumns("p1")},
-            ${aliasedPersonColumns("p2")}
+            ${aliasedPersonColumns("p2")},
+            fp1.create_type,
+            fp1.created_at,
+            fp1.created_by,
+            fp1.modify_type,
+            fp1.modified_at,
+            fp1.modified_by,
+            fp1.created_from_application
         FROM new_fridge_partner fp1
         JOIN new_fridge_partner fp2 ON fp1.partnership_id = fp2.partnership_id AND fp1.indx = 1 AND fp2.indx = 2
         JOIN person p1 ON fp1.person_id = p1.id
         JOIN person p2 ON fp2.person_id = p2.id
         """
-            .trimIndent()
-
-    return createQuery(sql)
-        .bind("partnershipId", UUID.randomUUID())
-        .bind("person1", personId1)
-        .bind("person2", personId2)
-        .bind("startDate", startDate)
-        .bind("endDate", endDate)
-        .bind("conflict", conflict)
+            )
+        }
         .exactlyOne(toPartnership("p1", "p2"))
 }
 
 fun Database.Transaction.updatePartnershipDuration(
     id: PartnershipId,
     startDate: LocalDate,
-    endDate: LocalDate?
+    endDate: LocalDate?,
+    modifyType: ModifyType,
+    modifiedAt: HelsinkiDateTime,
+    modifiedBy: EvakaUserId?
 ): Boolean {
-    // language=SQL
-    val sql =
-        """
-        UPDATE fridge_partner SET start_date = :startDate, end_date = :endDate
-        WHERE partnership_id = :id
+    return createQuery<Any> {
+            sql(
+                """
+        UPDATE fridge_partner SET start_date = ${bind(startDate)}, end_date = ${bind(endDate)}, modify_type = ${bind(modifyType)}, modified_at = ${bind(modifiedAt)}, modified_by = ${bind(modifiedBy)}
+        WHERE partnership_id = ${bind(id)}
         RETURNING partnership_id
         """
-            .trimIndent()
-
-    return createQuery(sql)
-        .bind("id", id)
-        .bind("startDate", startDate)
-        .bind("endDate", endDate)
+            )
+        }
         .mapTo<PartnershipId>()
         .useIterable { it.firstOrNull() } != null
 }
 
-fun Database.Transaction.retryPartnership(id: PartnershipId) {
-    // language=SQL
-    val sql = "UPDATE fridge_partner SET conflict = false WHERE partnership_id = :id"
-
-    createUpdate(sql).bind("id", id).execute()
+fun Database.Transaction.retryPartnership(
+    id: PartnershipId,
+    modifiedById: EvakaUserId?,
```

Comment:
`modifiedById` is always given and shouldn't be nullable to avoid accidental mistakes in the calling side.